op-preimage/iface.go

```diff
@@ -62,6 +68,40 @@ func (k Keccak256Key) TerminalString() string {
 	return "0x" + hex.EncodeToString(k[:])
 }
 
+// Sha256Key wraps a sha256 hash to use it as a typed pre-image key.hash
+type Sha256Key [32]byte
+
+func (k Sha256Key) PreimageKey() (out [32]byte) {
+	out = k
+	out[0] = byte(Sha256KeyType)
+	return
+}
+
+func (k Sha256Key) String() string {
+	return "0x" + hex.EncodeToString(k[:])
+}
+
+func (k Sha256Key) TerminalString() string {
+	return "0x" + hex.EncodeToString(k[:])
+}
+
+// BlobKey is the hash of a blob commitment and `z` value to use as a preimage key for `y`.
+type BlobKey [32]byte
+
+func (k BlobKey) PreimageKey() (out [32]byte) {
+	out = k
+	out[0] = byte(BlobKeyType)
+	return
+}
+
+func (k BlobKey) String() string {
+	return "0x" + hex.EncodeToString(k[:])
+}
+
+func (k BlobKey) TerminalString() string {
+	return "0x" + hex.EncodeToString(k[:])
+}
```

Comment:
These methods are good candidates for unit tests. Let's add some to a new `iface_test.go` source file or pull them out into another file.